### PR TITLE
CTCP-4647: Bugs 4,6,7

### PR DESCRIPTION
**Pull request by BledarFeratiEY**

### CODE REVIEW

**Review by joecefnhedog — APPROVED** [review]

**Comment by BledarFeratiEY** [comment]
rebuild

**Comment by BledarFeratiEY** [comment]
rebuild

File: app/controllers/loading/LocationController.scala [hunk 1]
```diff
@@ -36,41 +38,61 @@ class LocationController @Inject() (
...
-        Ok(view(preparedForm, departureId, countryName, loadingLocationMaxLength, mode))
-    }
+  def onPageLoad(departureId: String, mode: Mode): Action[AnyContent] = actions.requireData(departureId).async {
+    implicit request =>
+      countriesService.getCountries().map {
+        countryList =>
+          val countryIE15 = request.userAnswers.departureData.Consignment.PlaceOfLoading.flatMap(
+            pol =>
+              pol.country match {
+                case Some(country) => countryList.values.find(_.code.code == country)
```
Comment (joecefnhedog) [inline app/controllers/loading/LocationController.scala:58]: can this just be a pol.country.map {..

**Review by joecefnhedog — DISMISSED** [review]

**Review by joecefnhedog — APPROVED** [review]

**Review by joecefnhedog — APPROVED** [review]
approved

**Review by justega247 — APPROVED** [review]

**Pull request merged**